plugins/topology/src/components/Graph/BaseNode.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+import {
+  BadgeLocation,
+  DEFAULT_LAYER,
+  DefaultNode,
+  Layer,
+  Node,
+  NodeStatus,
+  observer,
+  ScaleDetailsLevel,
+  TOP_LAYER,
+  useCombineRefs,
+  WithContextMenuProps,
+  WithDndDropProps,
+  WithDragNodeProps,
+  WithSelectionProps,
+  useHover,
+  WithCreateConnectorProps,
+} from '@patternfly/react-topology';
+import { getKindStringAndAbbreviation } from '../../utils/workload-node-utils';
+
+type BaseNodeProps = {
+  className?: string;
+  innerRadius?: number;
+  icon?: string;
+  kind?: string;
+  labelIcon?: React.ReactNode;
+  labelIconPadding?: number;
+  badge?: string;
+  badgeColor?: string;
+  badgeTextColor?: string;
+  badgeBorderColor?: string;
+  badgeClassName?: string;
+  badgeLocation?: BadgeLocation;
+  children?: React.ReactNode;
+  attachments?: React.ReactNode;
+  element: Node;
+  hoverRef?: (node: Element) => () => void;
+  dragging?: boolean;
+  dropTarget?: boolean;
+  canDrop?: boolean;
+  nodeStatus?: NodeStatus;
+  showStatusBackground?: boolean;
+  alertVariant?: NodeStatus;
+} & Partial<WithSelectionProps> &
+  Partial<WithDragNodeProps> &
+  Partial<WithDndDropProps> &
+  Partial<WithContextMenuProps> &
+  Partial<WithCreateConnectorProps>;
```

Comment:
```suggestion
  Partial<WithContextMenuProps>;
```
We don't need these props if we aren't allowing edge modifications.